forem/forem
# Prevent suspended users from self-deleting and returning

## What type of PR is this? (check all applicable)


## Description

Currently banned users can delete and recreate their account to get out of banned status. To avoid this, this PR checks on user delete if the user is currently banned and if so stores a hashed version of their username. On signup, we compare against this table and display an error message if someone is trying to sign up with a previously blocked username. 

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/19

## QA Instructions, Screenshots, Recordings

1. Sign up a new user via Github or Twitter
2. Ban the user, e.g.
    ```ruby
    user = User.find(...)
    user.add_role(:banned)
3. Delete the user via the service object: 
    ```ruby
    User::Delete.call(user)
4. Try to sign up again. It should NOT work and you'll see a message in the global notice flash area.

### UI accessibility concerns?

n/a

## Added tests?


## Added to documentation?
